src/main/scala/beam/sim/ArgumentsParser.scala

```diff
@@ -0,0 +1,104 @@
+        .action(
+          (value, args) => {
+            args.copy(
+              config = Some(BeamConfigUtils.parseFileSubstitutingInputDirectory(value)),
+              configLocation = Option(value)
+            )
+          }
+        )
+        .validate(
+          value =>
+            if (value.trim.isEmpty) failure("config location cannot be empty")
+            else success
+        )
+        .text("Location of the beam config file")
+      opt[String]("cluster-type")
+        .action(
+          (value, args) =>
+            args.copy(
+              clusterType = value.trim.toLowerCase match {
+                case "master" => Some(Master)
+                case "worker" => Some(Worker)
+                case _        => None
+              }
+          )
+        )
+        .text("If running as a cluster, specify master or worker")
+      opt[String]("node-host")
+        .action((value, args) => args.copy(nodeHost = Option(value)))
+        .validate(value => if (value.trim.isEmpty) failure("node-host cannot be empty") else success)
+        .text("Host used to run the remote actor system")
+      opt[String]("node-port")
+        .action((value, args) => args.copy(nodePort = Option(value)))
+        .validate(value => if (value.trim.isEmpty) failure("node-port cannot be empty") else success)
+        .text("Port used to run the remote actor system")
+      opt[String]("seed-address")
+        .action((value, args) => args.copy(seedAddress = Option(value)))
+        .validate(
+          value =>
+            if (value.trim.isEmpty) failure("seed-address cannot be empty")
+            else success
+        )
+        .text(
+          "Comma separated list of initial addresses used for the rest of the cluster to bootstrap"
+        )
+      opt[Boolean]("use-local-worker")
+        .action((value, args) => args.copy(useLocalWorker = Some(value)))
+        .text(
+          "Boolean determining whether to use a local worker. " +
+          "If cluster is NOT enabled this defaults to true and cannot be false. " +
+          "If cluster is specified then this defaults to false and must be explicitly set to true. " +
+          "NOTE: For cluster, this will ONLY be checked if cluster-type=master"
+        )
+
+      checkConfig(
+        args =>
+          if (args.useCluster && (args.nodeHost.isEmpty || args.nodePort.isEmpty || args.seedAddress.isEmpty))
+            failure("If using the cluster then node-host, node-port, and seed-address are required")
+          else if (args.useCluster && !args.useLocalWorker.getOrElse(true))
+            failure("If using the cluster then use-local-worker MUST be true (or unprovided)")
+          else success
+      )
+    }
+  }
+
+  def parseArguments(parser: OptionParser[Arguments], args: Array[String]): Option[Arguments] = {
+    parser.parse(args, init = Arguments())
+  }
+
+  def parseArguments(args: Array[String]): Option[Arguments] = parseArguments(buildParser, args)
```

Comment:
This seems to be the only public method. Maybe we should hide the rest unless it becomes necessary to expose it?

Reply:
done